src/plugins/intel_gna/gna_plugin.cpp

```diff
@@ -1001,6 +1001,18 @@ void GNAPlugin::LoadNetwork(const CNNNetwork& _network) {
                                 << " after compiling GNA graph";
         }
         portId++;
+
+        if (LayerInfo(outLayer).isOutput()) {
+            auto component = graphCompiler.dnnComponents.findComponent(outLayer);
+
+            auto dims = outputs_.at(outPort.second->getName()).dims;
+            if (component && dims.size() >= 2 && component->num_rows_out == dims[0] &&
+                component->num_columns_out == dims[1]) {
```

Comment:
What about other dimensions for case when dimensions size is greater than 2?

Reply:
I think we should tighten the condition to 
dims.size() == 2
dims[0] != dims[1]
component.type == Interleave or deinterleave

Reply:
The change is only for input data which are loaded to input layers by ImportFrames in QueueInference and output data which are exported by by ExportScores in WaitFor.
It looks like only first dimensions really matter if transposition is done or note due the fact all other elements are copied as one big linear chunk of data.

In my opinion this affects input and output layers only so I suspect it does not matter if condition is that dimension is equal two or grater than 2
dims.size() == 2

This is why I am not fully convinced to  add This to condition:
dims[0] != dims[1]
component.type == Interleave or deinterleave
due the fact they will apply only to the situation in the ticket https://jira.devtools.intel.com/browse/CVS-88521 but not resolve the issue of loading and exporting.

I will add functional tests and try to cover problematic cases.

Reply:
can we reevaluate this comment?